### PR TITLE
initial integration tests

### PR DESCRIPTION
**Pull request by Cyrik**
## What has Changed?

- added integration for nrepl start
  - loads a clj file
  - starts a deps.edn nrepl
  - evals the file
  - checks for calva-output
- switches docker image to one that has Clojure installed

We can use this as a base for more integration tests, but it's a good smoke test for now.

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tests
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
     - [x] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [x] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Created the issue I am fixing/addressing, if it was not present.

Ping @pez, @bpringe

### CODE REVIEW

**Comment by Cyrik** [comment]
still very much playing around, but wanted to share my initial working version

File: .vscode/launch.json [hunk 1]
```diff
@@ -142,7 +142,7 @@
             "stopOnEntry": false,
             "sourceMaps": true,
             "env": {
-                "IS_DEBUG": "true",
+            "IS_DEBUG": "true",
```
Comment (PEZ) [inline .vscode/launch.json]: This looks a bit odd?

File: src/extension-test/integration/suite/extension-test.ts [hunk 2]
```diff
@@ -23,12 +27,80 @@ suite('Extension Test Suite', () => {
...
+    // test("open a file and close it again, w/o croaking", async () => {
+    //   expect(vscode.window.activeTextEditor).undefined;
+    //   const testClj = await openFile(path.join(util.testDataDir, 'test.clj'));
+    //   vscode.commands.executeCommand("workbench.action.closeActiveEditor");
+    //   await sleep(1000);
+    //   expect(vscode.window.activeTextEditor).undefined;
+    // });
+
+    test('connect to repl', async function () {
+        console.log('connect to repl');
```
Comment (PEZ) [inline src/extension-test/integration/suite/extension-test.ts]: This probably should be _'start repl and connect (jack-in)'_ as _connect_ has a the special meaning in Calva of assuming the REPL is already started.

**Review by PEZ — CHANGES_REQUESTED** [review]
This is wonderful!!

The name of the test is the only thing I'd like fixed before merging.

File: .vscode/launch.json (hunk 1 above)
Comment (Cyrik) [reply .vscode/launch.json]: ah whoops, that was an accident, good catch.

Comment (PEZ) [reply .vscode/launch.json]: Isn't it strange that it passed the prettier test step, @corasaurus-hex ?

**Review by PEZ — APPROVED** [review]
🙏 ❤️ 🎉

File: .vscode/launch.json (hunk 1 above)
Comment (corasaurus-hex) [reply .vscode/launch.json]: Prettier just formats typescript and javascript right now. https://github.com/BetterThanTomorrow/calva/blob/published/package.json#L2584-L2585

We could expand it to also format json and other file types, though!

**Pull request merged**